glommio/src/executor/mod.rs

```diff
@@ -572,75 +579,145 @@ impl Default for LocalExecutorBuilder {
 
 /// Specifies a policy by which [`LocalExecutorPoolBuilder`] distributes
 /// [`LocalExecutor`]s on the machine's CPUs / hardware topology.
+///
+/// The default is `Placement::Unbound`.
+///
+/// ## Example
+///
+/// Some `Placement`s allow manually filtering available CPUs via a [`CpuSet`],
+/// such as `MaxSpreadOnCpus`.  The following would place shards on four CPUs
+/// (a.k.a.  hyper-threads) that are on numa node 0 and have an even numbered
+/// package ID according to their [`CpuLocation`]. The selection aims to achieve
+/// a high degree of separation between the CPUs in terms of machine topology.
+/// Each [`LocalExecutor`] would be bound to a single CPU.
+///
+/// Note that if four CPUs are not available, the call to
+/// [`LocalExecutorPoolBuilder::on_all_shards`] would return an `Err` when using
+/// `MaxSpreadOnCpus`.
+///
+/// ```
+/// use glommio::{CpuSet, LocalExecutorPoolBuilder, Placement};
+///
+/// let cpus = CpuSet::online()
+///     .expect("Err: please file an issue with glommio")
+///     .filter(|l| l.numa_node == 0)
+///     .filter(|l| l.package % 2 == 0);
+///
+/// let handles = LocalExecutorPoolBuilder::new(4)
+///     .placement(Placement::MaxSpreadOnCpus(cpus))
+///     .on_all_shards(|| async move {
+///         // ... important stuff ...
+///     })
+///     .unwrap();
+///
+/// handles.join_all();
+/// ```
 #[derive(Debug)]
 pub enum Placement {
     /// For the `Unbound` variant, the [`LocalExecutor`]s created by a
-    /// [`LocalExecutorPoolBuilder`] are not bound to any CPU.
+    /// [`LocalExecutorPoolBuilder`] are not bound to any CPU.  This is the
+    /// default placement.
     Unbound,
-    /* MaxSpread,
-     * MaxPack,
+    /// The `SharedOnCpus` variant binds each [`LocalExecutor`] to the set of
+    /// CPUs specified by [`CpuSet`].  With an unfiltered CPU
+    /// set returned by [`CpuSet::online`], this is similar to using `Unbound`
+    /// with the distinction that bringing additional CPUs online will not
+    /// allow the executors to run on the newly available CPUs.  The
+    /// `SharedOnCpus` variant allows the number of shards specified in
+    /// [`LocalExecutorPoolBuilder::new`] to be greater than the number of CPUs
+    /// as long as at least one CPU is included in `CpuSet`.
+    ///
+    /// #### Errors
+    ///
+    /// If the provided [`CpuSet`] contains no CPUs, a call to
+    /// [`LocalExecutorPoolBuilder::on_all_shards`] will return `Result::
+    /// Err`.
+    SharedOnCpus(CpuSet),
```

Comment:
this is a bit inconsistent with the rest

While you have `MaxPack` and `MaxPackOnCpus`, the `OnCpus` version of `Unbound` is... `Shared`?